undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentInfoService.java

```diff
@@ -982,6 +983,10 @@ private DeploymentInfo createServletConfig() throws StartException {
                 webSocketDeploymentInfo.setBuffers(servletContainer.getWebsocketsBufferPool().getValue());
                 webSocketDeploymentInfo.setWorker(servletContainer.getWebsocketsWorker().getValue());
                 webSocketDeploymentInfo.setDispatchToWorkerThread(servletContainer.isDispatchWebsocketInvocationToWorker());
+
+                // Enables compression. This fixes WFLY-6787
+                webSocketDeploymentInfo.addExtension(new PerMessageDeflateHandshake(false));
```

Comment:
Just a note on this: In EAP 6.4 this was not configurable behavior, it was always on. So I agree it needs to be configurable, but adding it non-configurable is a true restoration of previous behavior, not just a partial one.

The basic question here is whether having it always on in 10.1 is acceptable (e.g. perf impact or something) or whether this should just wait for 11 and be done with the config option.